src/test/java/edu/ucla/library/iiif/fester/handlers/PostCsvHandlerTest.java

```diff
@@ -261,6 +264,74 @@ public final void testWorksCSV(final TestContext aContext) throws IOException, C
         });
     }
 
+    /**
+     * Ensures that uploaded files are deleted.
+     *
+     * @param aContext A test context
+     */
+    @Test
+    public final void testDeleteUploadedFilesOnEnd(final TestContext aContext) throws IOException, CsvException {
+        final int port = aContext.get(Config.HTTP_PORT);
+        final WebClient webClient = WebClient.create(myVertx);
+        final HttpRequest<Buffer> postRequest = webClient.post(port, UNSPECIFIED_HOST, COLLECTIONS_PATH);
+        final String filePath = FULL_CSV_FILE.getAbsolutePath();
+        final String fileName = FULL_CSV_FILE.getName();
+        final MultipartForm form = MultipartForm.create();
+        final List<String[]> expected = LinkUtilsTest.read(FULL_CSV_FILE.getAbsolutePath());
+        final String host = StringUtils.format(HOST, port);
+        final Async asyncTask = aContext.async();
+
+        form.textFileUpload(Constants.CSV_FILE, fileName, filePath, Constants.CSV_MEDIA_TYPE);
+
+        postRequest.sendMultipartForm(form, postHandler -> {
+            if (postHandler.succeeded()) {
+                final HttpResponse<Buffer> postResponse = postHandler.result();
+                final String postStatusMessage = postResponse.statusMessage();
+                final int postStatusCode = postResponse.statusCode();
+
+                if (postStatusCode == HTTP.CREATED) {
+                    final Buffer actual = postResponse.body();
+                    final String contentType = postResponse.getHeader(Constants.CONTENT_TYPE);
+
+                    // Check that what we get back is the same as what we sent
+                    check(aContext, LinkUtils.addManifests(host, expected), actual);
+
+                    // Check that what we get back has the correct media type
+                    aContext.assertEquals(Constants.CSV_MEDIA_TYPE, contentType);
+
+                    // Check that the UUID-named CSV file was deleted
+                    final Matcher uploadedFilePathMatcher = Pattern
+                            .compile("(" + BodyHandler.DEFAULT_UPLOADS_DIRECTORY + "\\/" + "[0-9a-f\\-]+" + ")")
+                            .matcher(postStatusMessage);
+                    uploadedFilePathMatcher.find();
+                    final String uploadedFilePath = uploadedFilePathMatcher.group(1);
+
+                    // Wait for the file to get deleted; 500 ms should be long enough, unless the file is huge
+                    final Long timerDelay = 500L;
```

Comment:
This is just a stylistic thing that I'll bring up for your thoughts. I tend to like defining variables at the top of a block (keeping them in a predictable place). I know others like defining them just before they're used (in case for some reason they don't end up needing to be used). A compromise might be to put the `final Long timerDelay;` under `final String contentType ...` and to initialize it where it is now with `timerDelay = 500;`. It's just a stylistic preference, but bringing it up to see if you have strong feelings about any of the three options.